runtime/runtime/src/pipelining.rs

```diff
@@ -111,6 +118,7 @@ impl ReceiptPreparationPipeline {
     pub(crate) fn submit(
         &mut self,
         receipt: &Receipt,
+        state_update: &TrieUpdate,
```

Comment:
Can we keep the `TrieUpdate` isolated to within `schedule_contract_preparation` function call? I remember having a good reason to build `submit` the way it is (i.e. the somewhat clunky passing of `account` via `LazyCell`?)

And if that's too clunky, lets at least replace the `account` LazyCell with a read off `TrieUpdate` too.

Reply:
I'm not sure what was the reason for using `LazyCell` for account, my best guess is to cache it in order to avoid multiple trie lookups when `handle_receipt` is called recursively. This seems like a pretty rare case and trie reads are served from memory anyway, so I think it is OK not to cache it.
I've removed `account` argument and replace its usage with direct reads from `state_update`, could you please take another look.